docs/code_ref/coordinates/rotatedsunframe.rst

```diff
@@ -0,0 +1,384 @@
+The ``RotatedSun*`` frame can also be the destination frame, which can be more intuitive in some situations and even necessary in some others (due to API limitations).
+Let's use a coordinate from earlier, which represents the coordinate in a "real" coordinate frame::
+
+  >>> coord = rs_hgc.transform_to(rs_hgc.base)
+  >>> coord
+  <HeliographicCarrington Coordinate (obstime=2020-03-04T00:00:00.000): (lon, lat, radius) in (deg, deg, km)
+      (45.13354448, 20., 695700.)>
+
+If we create a ``RotatedSun*`` frame for a different base time, we can represent that same point using coordinates prior to differential rotation::
+
+  >>> rs_frame = RotatedSunFrame(base=f.HeliographicCarrington(obstime=coord.obstime), rotated_time="2020-03-06 12:00")
+  >>> rs_frame
+  <RotatedSunHeliographicCarrington Frame (base=<HeliographicCarrington Frame (obstime=2020-03-04T00:00:00.000)>, duration=2.5 d, rotation_model=howard)>
+
+  >>> new_coord = coord.transform_to(rs_frame)
+  >>> new_coord
+  <RotatedSunHeliographicCarrington Coordinate (base=<HeliographicCarrington Frame (obstime=2020-03-04T00:00:00.000)>, duration=2.5 d, rotation_model=howard): (lon, lat, radius) in (deg, deg, km)
+      (10., 20., 695700.)>
+
+There coordinates are stored in the ``RotatedSun*`` frame, but it can be useful to "pop off" this extra layer and retain only the coordinate representation in the base coordinate frame.
+There is a convenience method called `~sunpy.coordinates.metaframes.RotatedSunFrame.as_base()` to do exactly that.
+Be aware the resulting coordinate does *not* point to the same location in inertial space, despite the superficial similarity.
+Essentially, the component values have been copied from one coordinate frame to a different coordinate frame, and thus this is not merely a transformation between coordinate frames::
+
+  >>> new_coord.as_base()
+  <HeliographicCarrington Coordinate (obstime=2020-03-04T00:00:00.000): (lon, lat, radius) in (deg, deg, km)
+      (10., 20., 695700.)>
+
+Advanced uses of RotatedSunFrame
+--------------------------------
+
+Here are advanced examples of how to use ``RotatedSunFrame``.
+
+Plotting differentially rotated coordinates
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Coordinates in a ``RotatedSun*`` frame can be plotted onto a map in the same manner as a normal coordinate.
+Here, we plot the differentially rotated locations of a solar feature from -5 days to +5 days.
+
+.. plot::
+   :include-source:
+
+   from astropy.coordinates import SkyCoord
+   import astropy.units as u
+   import matplotlib.pyplot as plt
+
+   from sunpy.coordinates import RotatedSunFrame
+   from sunpy.data.sample import AIA_171_IMAGE
+   from sunpy.map import Map
+
+   m = Map(AIA_171_IMAGE)
+
+   fig = plt.figure()
+   ax = plt.subplot(projection=m)
+   m.plot(clip_interval=(1., 99.95)*u.percent)
+
+   durations = range(-5, 6, 1)*u.day
+   point = SkyCoord([187]*len(durations)*u.arcsec, [283]*len(durations)*u.arcsec, frame=m.coordinate_frame)
```

Comment:
The `*len(durations)` here is because of that astropy bug which stops the broadcasting working on obstime? 

I am wondering why we need to do that, and why specifying an array of `durations` for a singular coordinate point doesn't "just work".

Reply:
Yeah, I can make this easier...

Reply:
Okay, I've added automatic upgrading of the coordinate data, which makes the `plot_coord` example cleaner.